third_party/amd/lib/TritonAMDGPUToLLVM/Utility.cpp

```diff
@@ -122,6 +122,66 @@ Value shflIdxSync(Location loc, ConversionPatternRewriter &rewriter, Value val,
                         i32_val(0x1f));
 }
 
+Value storeShared(ConversionPatternRewriter &rewriter, Location loc, Value ptr,
+                  Value val, Value pred) {
+  auto ty = val.getType();
+  auto val_ty = vec_ty(ty, 1);
+  Value vec_val = undef(val_ty);
+  vec_val = insert_element(val_ty, vec_val, val, i32_val(0));
+
+  auto vec_ty = vec_ty(i1_ty, 1);
```

Comment:
Both vec_ty and val_ty are vector types, so I'm not sure this is the best name.  It seems like these are val_ty and pred_ty?